code/modules/unit_tests/mapping.dm

```diff
@@ -17,3 +17,61 @@
 			continue
 
 		TEST_FAIL(log_entry)
+
+/// Verifies that there are no space turfs inside a station area, or on any planetary z-level. Sometimes, these are introduced during the load of the map and are not present in the DMM itself.
+/// Let's just make sure that we have a stop-gap measure in place to catch these if they pop up so we don't put it onto production servers should something errant come up.
+/datum/unit_test/mapload_space_verification
+	// This test is quite taxing time-wise, so let's run it later than other faster tests.
+	priority = TEST_LONGER
+
+/datum/unit_test/mapload_space_verification/Run()
+	// Get the datum associated with our currently running map.
+	var/datum/map_config/current_map = SSmapping.config
+
+	// Is our current map a planetary station (NO space turfs allowed)? If so, check for ANY space turfs.
+	if(current_map.planetary)
+		validate_planetary_map()
+		return
+
+	// Let's explicitly outline valid areas to catch anything that we're... okay with for now, because there are some valid use cases (though we should try to avoid them).
+	var/list/excluded_area_typecache = typecacheof(list(
+		// Space! This is likely an intentional space turf, so let's not worry about it.
+		/area/space,
+		// Transit areas have space turfs in a valid placement.
+		/area/shuttle/transit,
+		// Space Ruins do their own thing (dilapidated lattices over space turfs, for instance). Rather than fuss over it, let's just let it through.
+		/area/ruin/space,
+		// Same stipulation as space ruins, but they're (ruined) shuttles instead.
+		/area/shuttle/ruin,
+		// Solars have lattices over space turfs, and are a valid placement for space turfs in a station area.
+		/area/station/solars,
+	))
+
+	// We aren't planetary, so let's check area placements and ensure stuff lines up.
+	for(var/turf/iterated_turf in world)
```

Comment:
@LemonInTheDark Should this be `ALL_TURFS()`?

Reply:
using ALL_TURFS seems to make it fail alright:

```txt
[2022-11-21 01:31:24.967] Starting up round ID .
 - -------------------------
[2022-11-21 01:32:59.306] 	FAILURE #1: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (109,118,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #2: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (110,118,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #3: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (109,119,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #4: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (110,119,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #5: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (109,120,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
 - 	FAILURE #6: Space turf found in non-allowed area (/area/station/maintenance/tram/mid) at Central Tram Underpass (110,120,2)! Please ensure that all space turfs are in an /area/space! at code/modules/unit_tests/mapload_space_verification.dm:38
```